execution/execution.go

```diff
@@ -34,6 +35,22 @@ type ShuffleData struct {
 	Variables octosql.Variables
 }
 
+func mergeNextShuffles(first, second map[string]ShuffleData) (map[string]ShuffleData, error) {
+	result := first
+
+	for key, value := range second {
+		alreadyValue, ok := first[key]
+
+		if ok && !reflect.DeepEqual(alreadyValue, value) {
```

Comment:
reflect.DeepEqual shouldn't really be used in such cases. Just omit this check and do result[key] = value imo.